executor/builder.go

```diff
@@ -820,12 +821,81 @@ func (b *executorBuilder) buildHashJoin(v *plan.PhysicalHashJoin) Executor {
 	return e
 }
 
+// buildProjBelowAgg builds a ProjectionExec below AggregationExec.
+// If all the args of `aggFuncs`, and all the item of `groupByItems`
+// are columns or constants, we do not need to build the `proj`.
+func (b *executorBuilder) buildProjBelowAgg(aggFuncs []*aggregation.AggFuncDesc, groupByItems []expression.Expression, src Executor) Executor {
```

Comment:
we can use `v *plan.PhysicalHashAgg` to replace the first two parameters, which helps us to reduce the length of function signature.

Reply:
This is also used for PhysicalStreamAgg,
so pass two parameters may avoid the cost of type assertion.

Reply:
This is also used for PhysicalStreamAgg,
so pass two parameters may avoid the cost of type assertion.